workflow/executor/docker/docker.go

```diff
@@ -215,13 +215,23 @@ func (d *DockerExecutor) syncContainerIDs(ctx context.Context, containerNames []
 				return err
 			}
 			for _, l := range strings.Split(string(output), "\n") {
-				parts := strings.Split(strings.TrimSpace(l), "=")
-				if len(parts) != 2 {
+				parts := strings.Split(strings.TrimSpace(l), "|")
+				if len(parts) != 3 {
 					continue
 				}
-				containerName := parts[0]
-				containerID := parts[1]
+				status := strings.SplitN(parts[0], " ", 2)[0] // Created,Exited,Up,
+				containerName := parts[1]
+				containerID := parts[2]
 				if d.containers[containerName] == "" && containerID != "" {
+					if status == "Created" { // for "Created" we must check to see if it was an early (non-zero) exit
```

Comment:
Consider adding a test that exercises this case?

Reply:
I think this only occurs under load. The test case is to run the stress tests!

Reply:
Sounds good. So looks like #5252 isn't a regression since this case was not covered in previous releases anyways, right?

Reply:
I'm not actually clear, but I'm prioritizing like a regression